gammapy/datasets/simulate.py

```diff
@@ -70,25 +79,58 @@ def _evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
 
         Returns
         -------
-        npred : `~gammapy.maps.Map`
+        npred : `~gammapy.maps.RegionNDMap`
             Npred map.
         """
-        energy_axis = MapAxis.from_edges(
-            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        energy_true = dataset.edisp.edisp_map.geom.axes["energy_true"]
+        energy_new = energy_true.upsample(10)
+        target = evaluator.model.spatial_model.position
+        edisp = dataset.edisp.get_edisp_kernel(
+            dataset.geoms["geom"].axes["energy"], position=target
         )
+        region_exposure = dataset.exposure.to_region_nd_map(target)
+
+        if not time_axis:
+            tstart = dataset.gti.time_start
+            tstop = dataset.gti.time_stop
+            nbin = int(((tstop - tstart) / t_delta).to(""))
+            time_axis_eval = TimeMapAxis.from_time_bounds(
```

Comment:
This does not seem right to me:
- `tstart` is an array of times, while `.from_time_bounds` would require a single value.
- You can maybe use https://github.com/gammapy/gammapy/blob/main/gammapy/maps/axes.py#L2734 instead

Reply:
If I'm not wrong, `from_gti` creates a `TimeMapAxis` with a number of time bins equal to the number of `tstart` and `tstop` of the GTIs. However, this is not what we need in the event sampler... 
`tstart` and `tstop` are arrays of times in general, but the datasets we use for the simulator usually store only one value. Do we expect to simulate from a dataset with gaps in its GTIs? I suppose this functionality is not implemented yet in the event sampler..